gemini/use-cases/education/resume-generation/resume_generate.ipynb

```diff
@@ -0,0 +1,979 @@
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "a176f957-bbf2-4d4c-9ab6-82d6f691adfa"
+      },
+      "outputs": [],
+      "source": [
+        "# Resume structure\n",
+        "resume_struc = [\n",
+        "    \"Personal_Information\",\n",
+        "    \"Professional_Summary\",\n",
+        "    \"Education\",\n",
+        "    \"Coursework\",\n",
+        "    \"Skills\",\n",
+        "    \"Experience\",\n",
+        "    \"Certifications\",\n",
+        "    \"Languages\",\n",
+        "    \"Extracurricular_Activities\",\n",
+        "    \"References\",\n",
+        "]"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qV1pqxYgSFr0"
+      },
+      "source": [
+        "### Defining the core functionalities\n",
+        "\n",
+        "This code defines a series of functions for processing resumes. It uses Gemini to extract key information like contact details and to generate tailored professional summaries. The code also includes functions for parsing resumes from different file formats (docx, pdf, txt), handling potential missing information, and structuring the final output for readability.\n",
+        "\n",
+        "*   def check_info(resume: str) -> Dict[str, str]: Uses Gemini to extract name, address, mobile number, and email from a resume, returning 'Not Found' if information is missing.\n",
+        "    \n",
+        "*   def get_info(resume: str) -> str: Similar to check_info but returns the extracted information as a formatted string.\n",
+        "\n",
+        "*   def iter_block_items(parent: Table) -> str:\n",
+        "    Iterates through a docx file object, yielding paragraphs and tables.\n",
+        "\n",
+        "*   def convert_docx_to_text(file_path: str) -> str:\n",
+        "    Parses a docx file and converts it to text, handling tables and paragraphs.\n",
+        "\n",
+        "*   def parse_resume(res_path: str) -> str:\n",
+        "    Parses a resume from various file types (docx, pdf, txt) with fallbacks to handle missing personal information.\n",
+        "\n",
+        "*   def get_response(parsed_res: str, parsed_jd: str) -> str:\n",
+        "    Generates a resume using Gemini, incorporating extracted information, a professional summary, and arranged skills/coursework.\n",
+        "\n",
+        "*   def structure_response(resume_generated: str) -> str:\n",
+        "    Structures the generated resume into a specific format for readability."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "c8dc8fe8-890c-4471-915a-396787e84a1d"
+      },
+      "outputs": [],
+      "source": [
+        "def check_info(resume: str) -> Dict[str, str]:\n",
+        "\n",
+        "    vertexai.init(\n",
+        "        project=\"aniket-tdcx\", location=\"us-central1\"\n",
+        "    )  # Initialize the AI Platform client with project ID\n",
```

Comment:
This initialization should only be done once at the beginning. And remove your project ID.